mkdocs.yml

```diff
@@ -26,14 +26,15 @@ pages:
   - 'Introduction': index.md
   - 'Motivation': motivation.md
   - 'Architecture': architecture.md
-  - 'Getting Started': user-guide.md
-  - 'User Guide':
-    - 'Creating Nix builds for your projects': user-guide/projects.md
-    - 'Stack Projects': user-guide/stack-projects.md
-    - 'Cabal Projects': user-guide/cabal-projects.md
-    - 'Hackage and Stackage': user-guide/hackage-and-stackage.md
-    - 'How to create a development environment': user-guide/development.md
-    - 'How to cross-compile your project': user-guide/cross-compilation.md
+  - 'Tutorials':
+    - 'Getting Started': tutorials/getting-started.md
+    - 'Creating a development environment': tutorials/development.md
+    - 'Sourcing files only part of git repository using cleanGit': tutorials/clean-git.md
+    - 'Mapping non-Haskell dependencies to Nixpkgs': tutorials/pkg-map.md
+    - 'Bumping Hackage and Stackage snapshots': tutorials/hackage-stackage.md
+    - 'Cross-compiling your project': tutorials/cross-compilation.md
+    - 'Installing nix-tools': tutorials/installing-nix-tools.md
```

Comment:
I also don't think users should usually do this. You can do materialization and everything without ever doing this. Another advanced/troubleshooting topic IMO.

Reply:
That's why I moved it to a separate tutorial. I wouldn't split tutorials into basic/advanced, but rather explain in the tutorial itself when something is needed and what's the goal.

Reply:
So you're saying that manually generating expressions is never really recommended anymore and materialized is the new way to go avoiding IFD?

Reply:
Not sure I can answer that fully, but at least you can get fully materialized Nix files for your project without ever installing `nix-tools`, just by running specific scripts to generate the files (which of course use `nix-tools` under the hood).

Reply:
@angerman what do you think? It would be great if we can communicate that in the documentation.

Either way I think having docs to install nix-tools is useful, but maybe they can be moved to ceveloper section.

Reply:
Can we change the "Manually generating Nix expressions" tutorial so that it shows how to use the materialization scriptt? Then the info about installing and using nix-tools and can be put in a dev note.

Reply:
:rocket:

Reply:
The whole materialisation mechanism is there to supersede the need for manual generation of expression with nix tools and doing so in a principled way that reduces potential pitfalls.

Conceptually pre-generating nix expressions is a form of caching. Caching is hard. Materialisation is a form of making sure haskell.nix generates the right nix expression you want in the first place. And then telling you how to copy that "cached" value out, and use it instead of recomputing.

Generally it is not a bad idea to understand how this all works the very basic level; this will provide intuition for this, and understanding how things fit together. My general approach when teaching concepts is to start at the very basic labour intensive setup. Then do the same thing again but use one level of abstraction; and finally do it all over again but this time use yet another layer of abstraction. This should help the person learning it to understand how the abstractions try to group together functionality and produce shorthands, that the student would have hopefully arrived at on his own as well. Now finding the right abstractions is *really* hard. Have we found them with haskell.nix? I'm not sure... only time will tell.

Thus, yes the recommended way is: IFD > materialisation > custom regeneration script > some custom setup that sits on top of what nix-tools generates.

The educational route is precisely the opposite way.

🤷‍♂️

Reply:
Most people won't want to understand it, but get things done with it.

I think that's a much bigger audience than those wanting to fiddle, so those docs should exist.

I would propose to address materialization vs manual expression in a separate PR after this one is merged.